### DISCUSSION TITLE
About Batch normalization

### DISCUSSION THREAD

**Discussion by skiyohara** [e3nn/e3nn#473] [Q&A]
Dear e3nn developers and users,

I am new to e3nn and currently working on coding a Convolutional Equivariant Neural Network.

In my toy data, I noticed that the tensor products produce comparatively larger values than typical convolutional networks. As a result, batch normalization is necessary to perform multiple convolutions effectively. In e3nn.nn.BatchNorm, the running_var (presumably representing variance in normalization) is multiplied by each field (Batch, n_features * (2l+1) tensor).

While I understand that this normalization operates similarly to the nonlinearity in e3nn, this multiplication increases the tensor outputs quite rapidly. Instead, I think it might be better for each field to be **divided** by running_var (norm_field).

Is my understanding correct?

Alternatively, are there methods to prevent tensor product values from becoming excessively large? e.g., normalization arg in TensorProduct?
Default setting in TensorProduct(path_normalization = "element", irrep_normalization = "component") was already used in my case.

Any comment will be helpful.

Shin
